src/averageAges.js

```diff
@@ -55,7 +72,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f' && (
+    ({ name }) => people.some(({ mother }) => mother === name)));
+
+  const children = people.filter(
+    ({ mother, sex }) => {
+      return mothers.some(
+        ({ name }) => mother === name) && (!onlyWithSon || sex === 'm');
+    });
+
+  const find = children.map(child => {
+    const foundMom = mothers.find(mom => mom.name === child.mother);
+
+    return foundMom ? child.born - foundMom.born : null;
+  }).filter(value => value);
+
+  const difference = find.reduce(
```

Comment:
```suggestion
  const ageDifference = find.reduce(
```

use more descriptive names